extensions/ql-vscode/src/authentication.ts

```diff
@@ -0,0 +1,60 @@
+import * as vscode from 'vscode';
+import * as Octokit from '@octokit/rest';
+
+const GITHUB_AUTH_PROVIDER_ID = 'github';
+
+// 'repo' scope should be enough for triggering workflows. For a comprenhensive list, see:
+// https://docs.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps
+const SCOPES = ['repo'];
+
+/** 
+ * Handles authentication to GitHub, using the VS Code [authentication API](https://code.visualstudio.com/api/references/vscode-api#authentication).
+ */
+export class Credentials {
+  private octokit: Octokit.Octokit | undefined;
+
+  async initialize(context: vscode.ExtensionContext): Promise<void> {
```

Comment:
By having a separate initialisation call after construction, it means there two things that the caller has to do, and if a developer doesn't read and understand how to use the `Credentials` class they could make one and forget to initialise it.

Would it be safer to do something like
```typescript
private constructor() {}

static async initialize(context: vscode.ExtensionContext): Promise<Credentials> {
  const c = new Credentials();
  c.registerListeners(context);
  c.setOctokit();
  return c;
}
```

and then to use it just do
```typescript
const credentials = await Credentials.initialize(ctx);
```

Unfortunately you can't do `async` constructors, but I did learn today you can do `static` methods like this so at least it looks a bit better / more understandable.